homeassistant/components/hive/__init__.py

```diff
@@ -156,22 +170,18 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
-class HiveSession(Entity):
-    """Initiate Hive Session Class."""
+class HiveEntity(Entity):
+    """Initiate Hive Base Class."""
 
-    entity_lookup = {}
-    core = None
-    heating = None
-    hotwater = None
-    light = None
-    sensor = None
-    switch = None
-    weather = None
-    attributes = None
+    def __init__(self):
```

Comment:
Accept the corresponding parameters for the instance attributes here. Then we don't need to set the attributes both in parent and in child class, but just in the parent.

```py
def __init__(self, session, hive_device):
    self.node_id = hive_device["Hive_NodeID"]
    self.node_name = hive_device["Hive_NodeName"]
    self.device_type = hive_device["HA_DeviceType"]
    self.node_device_type = hive_device["Hive_DeviceType"]
    self.session = session
    self.attributes = {}
    self._unique_id = f"{self.node_id}-{self.device_type}"
```

```suggestion
    def __init__(self, session, hive_device):
```